### PR TITLE
[mlir][spirv] Replace hardcoded attribute strings with op methods Resolve #77627

### LINKED ISSUE #77627: [mlir][spirv] Replace hardcoded attribute strings with op methods
Due to historical reasons we have quite some hardcoded strings for attribute names used in parser/printer and serializer/deserializer. Now in ODS we actually generate op methods for getting the names of attributes. It's better to replace the hardcoded strings for consistency. Specicially:

- [ ] Replace parser/printer attribute strings [here](https://github.com/llvm/llvm-project/blob/main/mlir/lib/Dialect/SPIRV/IR/SPIRVParsingUtils.h#L23-L50). For example, `branch_weights` can be replaced with `getBranchWeightsAttrName().strref()`.
- [x] Check and replace such patterns in serializer too, e.g., [here](https://github.com/llvm/llvm-project/blob/main/mlir/lib/Target/SPIRV/Serialization/SerializeOps.cpp#L701-L743). Need to check all the files in serializer library. But don't need to change the autogenerated ones given those are consistent for sure due to autogen.
- [x] Do the same for deserializer.

### PR DESCRIPTION
**Pull request by SahilPatidar**
Resolve #77627

### CODE REVIEW

File: mlir/lib/Target/SPIRV/Deserialization/Deserializer.cpp [hunk 1]
```diff
@@ -215,11 +215,14 @@ spirv::Deserializer::processMemoryModel(ArrayRef<uint32_t> operands) {
   if (operands.size() != 2)
     return emitError(unknownLoc, "OpMemoryModel must have two operands");
 
+  StringRef addressing_model = module->getAddressingModelAttrName().strref();
```
Comment (joker-eph) [inline mlir/lib/Target/SPIRV/Deserialization/Deserializer.cpp:218]: Why StringRef instead of StringAttr?

Comment (SahilPatidar) [reply mlir/lib/Target/SPIRV/Deserialization/Deserializer.cpp:218]: should I alter this all across the place?

Comment (joker-eph) [reply mlir/lib/Target/SPIRV/Deserialization/Deserializer.cpp:218]: Yes

Comment (SahilPatidar) [reply mlir/lib/Target/SPIRV/Deserialization/Deserializer.cpp]: done

**Review by antiagainst — APPROVED** [review]
Thanks for the change! LGM in general; but there are a few naming style issues. I just went ahead and fixed them given that you have been waiting for a while. Sorry about the delay in reviewing!

**Pull request merged**